packages/strapi-admin/admin/src/containers/Users/ProtectedEditPage/index.js

```diff
@@ -16,13 +17,23 @@ const ProtectedEditPage = () => {
     isLoading,
     allowedActions: { canRead, canUpdate },
   } = useUserPermissions(permissions);
+  const { state } = useLocation();
+  const from = get(state, 'from', '/');
+
+  useEffect(() => {
+    if (!isLoading) {
+      if (!canRead && !canUpdate) {
+        strapi.notification.info('notification.permission.not-allowed-read');
+      }
+    }
```

Comment:
Why don't you use only 1 if ?

Reply:
Because I find it easier to read the steps in such hook. It's very useful when we need to debug.